src/main/java/com/microsoft/azure/spring/data/cosmosdb/repository/support/SimpleDocumentDbRepository.java

```diff
@@ -135,6 +135,11 @@ private PartitionKey createKey(String partitionKeyValue) {
     @Override
     public Optional<T> findById(ID id) {
         Assert.notNull(id, "id must not be null");
+
+        if (id instanceof String && !StringUtils.hasText((String) id)) {
```

Comment:
I think in another PR, `id` can also be `Integer` and `int` type? Not sure which one is correct. Is there any conflict?

Also, `String` and `Integer` are `Object` types, but `int` is primitive type, so we allow `int` type?

Reply:
Yeah, both int and Integer are allowed. Here only String may has empty concept. take care of this here.